call.go

```diff
@@ -27,6 +27,10 @@ import (
 //
 // All errors returned by Invoke are compatible with the status package.
 func (cc *ClientConn) Invoke(ctx context.Context, method string, args, reply interface{}, opts ...CallOption) error {
+	// allow interceptor to see all applicable call options, which means those
+	// configured as defaults from dial option as well as per-call options
+	opts = append(cc.dopts.callOptions, opts...)
```

Comment:
@jhump this looks like data race in case of concurrent access with non-empty `opts ...CallOptions` as they override the same elements in `callOptions` (given sufficient capacity)

Reply:
Good catch! See #1948